core/dbt/contracts/graph/nodes.py

```diff
@@ -271,16 +271,20 @@ def add_public_node(self, value: str):
 
 
 @dataclass
-class ParsedNodeMandatory(GraphNode, HasRelationMetadata, Replaceable):
+class RelationalNode(HasRelationMetadata):
```

Comment:
I think it's confusing to use Node in the name here. It's not a Node. These classes are already complicated enough, I don't think it's necessarily a good idea to reuse the existing classes in nodes.py.  You're actually reimplementing a Relation class.  Do you actually need the separate database/schema/identifier? ("identifier" is what it's actually called in the Relation classes, and makes more sent to me, at least.).  I'm not exactly sure how this is going to be used. Would a relation_name (clone_relation_name?) field work?